### PR TITLE
rm  create_savepoint, release_savepoint methods from AbstractAdapter

### PR DESCRIPTION
**Pull request by vipulnsward**
They have been moved to Savepoints module

EDIT:Added another change to move savepoints include to abstract adapter, which depends on previous change.

### CODE REVIEW

File: activerecord/lib/active_record/connection_adapters/abstract_adapter.rb [hunk 1]
```diff
@@ -220,11 +221,6 @@ def supports_bulk_alter?
         false
       end
 
-      # Does this adapter support savepoints?
-      def supports_savepoints?
-        false
-      end
```
Comment (vipulnsward) [inline activerecord/lib/active_record/connection_adapters/abstract_adapter.rb]: All adapters support savepoint right now.

File: activerecord/lib/active_record/connection_adapters/abstract/savepoints.rb [hunk 2]
```diff
@@ -1,6 +1,7 @@
 module ActiveRecord
   module ConnectionAdapters
-    module Savepoints #:nodoc:
+    module Savepoints
```
Comment (vipulnsward) [inline activerecord/lib/active_record/connection_adapters/abstract/savepoints.rb]: rm :nodoc: to maintain visibility of methods.

File: activerecord/lib/active_record/connection_adapters/abstract_adapter.rb (hunk 1 above)
Comment (jeremy) [inline activerecord/lib/active_record/connection_adapters/abstract_adapter.rb]: What about third party adapters?

Comment (vipulnsward) [inline activerecord/lib/active_record/connection_adapters/abstract_adapter.rb]: Hmm, makes sense. I see other `?` methods are false by default.

Comment (vipulnsward) [inline activerecord/lib/active_record/connection_adapters/abstract_adapter.rb]: Done.

**Comment by jeremy** [comment]
Squashed @ https://github.com/rails/rails/commit/35f32036b6782f8d7960c9636ee4b551dcd29f69

**Pull request closed without merging**